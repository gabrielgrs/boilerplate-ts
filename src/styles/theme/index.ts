export { default as colors } from './configs/colors'
export { default as sizes } from './configs/sizes'
