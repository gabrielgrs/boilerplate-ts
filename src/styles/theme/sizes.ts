const sizes = {
  xxs: '4px',
  xl: '8px',
}

export default sizes
