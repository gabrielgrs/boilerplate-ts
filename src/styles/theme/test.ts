import * as theme from './index'

describe('theme', () => {
  it('Theme should match snapshot', () => {
    expect(theme).toMatchSnapshot()
  })
})
