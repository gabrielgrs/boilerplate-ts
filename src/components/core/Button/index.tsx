import { Button as SButton } from './styles'
import { TButton } from './types'

function Button({ children }: TButton) {
  return <SButton>{children}</SButton>
}

export default Button
