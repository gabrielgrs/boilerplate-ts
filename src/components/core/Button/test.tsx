import { render, screen } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import * as theme from 'styles/theme'

import Button from './index'

describe('<Button />', () => {
  it('should render the medium size by default', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Button>I am a Button</Button>
      </ThemeProvider>
    )

    expect(screen.getByRole('button', { name: /I am a Button/i })).toHaveStyle({
      backgroundColor: '#0f0f0f',
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
