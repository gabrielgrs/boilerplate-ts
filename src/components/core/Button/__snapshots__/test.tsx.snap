// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the medium size by default 1`] = `
.c0 {
  background: #0f0f0f;
  color: #0f0f0f;
}

<button
  class="c0"
>
  I am a Button
</button>
`;
